tests/protocol/mocks.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2020 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+import contextlib
+import re
+from azurelinuxagent.common.protocol.wire import WireProtocol
+from azurelinuxagent.common.utils import restutil
+from tests.tools import patch
+from tests.protocol.mockwiredata import WireProtocolData
+
+
+@contextlib.contextmanager
+def mock_wire_protocol(mock_wire_data_file):
+    """
+    Creates a mock WireProtocol object that will return the data specified by 'mock_wire_data_file' (which must
+    follow the structure of the data files defined in tests/protocol/mockwiredata.py).
+
+    NOTE: This function creates mocks for azurelinuxagent.common.utils.restutil.http_request and
+          azurelinuxagent.common.protocol.wire.CryptUtil. These mocks can be stopped using the
+          methods stop_mock_http_get() and stop_mock_crypt_util().
+
+    The return value is an instance of WireProtocol augmented with these properties/methods:
+
+        * mock_wire_data - the WireProtocolData constructed from the mock_wire_data_file parameter.
+        * stop_mock_http_request() - stops the mock for restutil.http_request
+        * stop_mock_crypt_util() - stops the mock for CrypUtil
+        * stop() - stops both mocks
+    """
+    def stop_mock_http_request():
+        if stop_mock_http_request.mock is not None:
+            stop_mock_http_request.mock.stop()
+            stop_mock_http_request.mock = None
+    stop_mock_http_request.mock = None
+
+    def stop_mock_crypt_util():
+        if stop_mock_crypt_util.mock is not None:
+            stop_mock_crypt_util.mock.stop()
+            stop_mock_crypt_util.mock = None
+    stop_mock_crypt_util.mock = None
+
+    def stop():
```

Comment:
those stop functions are mainly for debugging purposes; they come handy when there are conflicts with the mocks used by tests and the ones used here.